### ISSUE TITLE
Feature request: Invert Selection

### ISSUE THREAD

**Issue by Taranchul** [Querz/mcaselector#136]
Hi there,
a function to invert a selection would be very handy. For example, you could select only the regions or chunks that you want to keep, and then invert the selection and delete everything else.

Many applications that work with selections have this, so it should be very intuitive for most users (IMHO more than to select everything first and then de-select what you want to keep).

Just a humble feature request for this already awesome tool. :)

**Comment by Querz** [comment]
This would be mostly a function for visual convenience, because the result of an "inverted" selection can be achieved with slightly more work:
* Inverting a selection when you want to *keep* all chunks in your selection can be done by exporting the selected chunks to a new empty folder, then manually deleting all mca files from the region folder and copying the exported files into the region folder.
* Inverting a selection resulting from a filter can be achieved by negating the filter. This requires a little bit of logical thinking though :) . There is actually some logic already in the code that would allow to negate the entire filter, but it's currently not used.

A huge problem with inverting a selection currently is that the entire selection is kept in memory at all time. So having a huge world with tens of thousands of region files could potentially cause memory problems when selecting a few chunks and then inverting the selection. This would be pretty easy to fix though, but would have to be integrate in all the other functions like exporting or deleting based on a selection.

In conclusion it's a nice idea and i'll most likely add it one day, but as you can achieve the same effect with minimal effort it's not at the top of the list :)

**Comment by Querz** [comment]
Added the ability to invert selections in the [last update](https://github.com/Querz/mcaselector/releases/latest). Let me know if something doesn't work as expected.

**Comment by Taranchul** [comment]
Sorry for the late answer. Thank you very much! 👍 I'll report any issues.

**Comment by Taranchul** [comment]
Sorry for necroing this issue, but I just stumbled over [this recent video tutorial](https://www.youtube.com/watch?v=1xx8fwynlRs&t=6m) that mentions the invert function at 6:00 minutes with an application example. Just to show you that your choice to include it was the right one. 😁 Thanks again!

And to fulfill my promise above, I report that I didn't encounter any issues with it, yet.

**Comment by Querz** [comment]
Oh nice, I haven't seen that one yet, thanks for sharing!

**Comment by Taranchul** [comment]
My pleasure, I found it when I looked for tutorials for someone else to whom I recommended your tool. 🙂
